packages/concerto-core/lib/decoratormanager.js

```diff
@@ -252,6 +252,186 @@ class DecoratorManager {
+                    'namespace':namespace
+                };
+                if (obj.declaration && obj.declaration!==''){
+                    target.declaration=obj.declaration;
+                }
+                if (obj.property && obj.property!==''){
+                    target.property=obj.property;
+                }
+                decos.forEach((dcs)=>{
+                    if (dcs.name!=='Term' && dcs.name!=='Term_description'){
+                        let decotatorObj={
+                            '$class': 'concerto.metamodel@1.0.0.Decorator',
+                            'name': dcs.name,
+                        };
+                        if (dcs.arguments){
+                            let args=dcs.arguments.map((arg)=>{
+                                return {
+                                    '$class':arg.$class,
+                                    'value':arg.value
+                                };
+                            });
+                            decotatorObj.arguments=args;
+                        }
+                        let dcsObject = {
+                            '$class': 'org.accordproject.decoratorcommands.Command',
+                            'type': 'UPSERT',
+                            'target': target,
+                            'decorator': decotatorObj,
+                        };
+                        dcsObjects.push(dcsObject);
+                    }
+                });
+            });
+            let dcmsForNamespace={
+                '$class': 'org.accordproject.decoratorcommands.DecoratorCommandSet',
+                'name': nameOfDcs,
+                'version': versionOfDcs,
+                'commands': dcsObjects
+            };
+            data.push(dcmsForNamespace);
+        });
+        return data;
+    }
+    /**
+     * Parses the dict data into an array of decorator jsons
+     * @param {Object} decoratorDict the input dict
+     * @param {String} locale locale for target vocab
+     * @returns {Array<Object>} the parsed decorator command set array
+     */
+    static parseVocabs(decoratorDict,locale){
+        let data = [];
+        Object.keys(decoratorDict).forEach((namespace)=>{
+            let strVoc='';
+            strVoc=strVoc+`locale: ${locale}\n`;
+            strVoc=strVoc+`namespace: ${namespace}\n`;
+            strVoc=strVoc+'declarations:\n';
+            let jsonData=decoratorDict[namespace];
+            let dictVoc={};
+            jsonData.forEach((obj)=>{
+                // check if obj.decl already in dictVoc
+                if (!dictVoc[obj.declaration]){
+                    dictVoc[obj.declaration]={
+                        propertyVocabs:{}
+                    };
+                }
+                let decos=JSON.parse(obj.dcs);
+                decos.forEach((dcs)=>{
+                    if (dcs.name==='Term' || dcs.name==='Term_description'){
```

Comment:
If I understand correctly, this function should handle any decorator that has the prefix `Term_`, right? For example, I should also be able to extract `Term_tooltip` or `Term_myCustomLabel` etc.

Reply:
@muskanbararia I agree with @mttrbrts 
For more reference: https://github.com/accordproject/concerto/blob/668a033d0f4df7aeddeaf42afc25b2588920148f/packages/concerto-vocabulary/lib/vocabularymanager.js#L326

https://github.com/accordproject/concerto/pull/664

Reply:
Updated the PR. I was incorrectly assuming that only `Term_description` is identifier for vocab